### DISCUSSION TITLE
Is the tiktok driver currently supported?

### DISCUSSION THREAD

**Discussion by gilbert-mojodojo** [SocialiteProviders/Providers#1044] [Q&A]
Is the driver for tiktok currently functional?

I have tried installing the driver as per the documentation in a similar way to Google and Facebook. 

The Google and Facebook driver worked as expected but the TikTok one resulted in the following error "Driver [tiktok] not supported."

**Comment by atymic** [comment]
Please provide a code sample.

**Reply by gilbert-mojodojo** [reply]
So the routing is done using the following

Route::get('/tiktok/auth/redirect/', function () {     
    return Socialite::driver('tiktok')->redirect();
});



I've also set up the services.php file as per the documentation with
'tiktok' => [
  'client_id' => env('TIKTOK_CLIENT_ID'),
  'client_secret' => env('TIKTOK_CLIENT_SECRET'),
  'redirect' => env('TIKTOK_REDIRECT_URI')
],


where the values are in the .env file

This is in addition to making the required change to the eventserviceprovider file
protected $listen = [
    \SocialiteProviders\Manager\SocialiteWasCalled::class => [
        // ... other providers
        \SocialiteProviders\TikTok\TikTokExtendSocialite::class.'@handle',
    ],
];
